mat64/shadow.go

```diff
@@ -0,0 +1,201 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mat64
+
+import "github.com/gonum/blas/blas64"
+
+const (
+	// regionOverlap is the panic string used for the general case
+	// of a matrix region overlap between a source and destination.
+	regionOverlap = "mat64: bad region: overlap"
+
+	// regionIdentity is the panic string used for the specific
+	// case of complete agreement between a source and a destination.
+	regionIdentity = "mat64: bad region: identical"
+
+	// mismatchedStrides is the panic string used for overlapping
+	// data slices with differing strides.
+	mismatchedStrides = "mat64: bad region: different strides"
+)
+
+// checkOverlap returns false if the receiver does not overlap data elements
+// referenced by the parameter and panics otherwise.
+//
+// checkOverlap methods return a boolean to allow the check call to be added to a
+// boolean expression, making use of short-circuit operators.
+
+func (m *Dense) checkOverlap(a blas64.General) bool {
```

Comment:
I see. So the idea is that it's the users problem if the types are different and there is incorrect overlap. That's probably reasonable since you have to explicitly construct two different matrices with the same underlying data. The one exception is RowView and ColView of Dense. 
````
v := m.ColView(1)
v.SolveVec(m, v)
````
This is an easy case to construct, but will be incorrect. I'm not exactly sure what the correct conclusion is.